### ISSUE TITLE
Timber\User `can` method does not match user_can function signature

### ISSUE THREAD

**Issue by kshaner** [timber/timber#2532]
The Timber\User::can method does not pass the arbitrary arguments to the WordPress core `user_can` function. This creates mismatched expectations for what the method can do.

## Expected behavior

I would expect that all Timber methods that are facades to WordPress core functions and use the same argument signatures.

## Actual behavior
Some capability checks always fail due to the lack of arguments. Certain capabilities that require arguments (edit_post, edit_user, etc) produces a PHP notice: 

```
Notice: Undefined offset: 0 in /var/www/wp-includes/capabilities.php on line 149
```

## Steps to reproduce behavior

This code does not work because `post.id` is not passed along to the `user_can` function.

```twig
{% if user.can( 'edit_post', post.id ) %}
    {# do something with privileges #}
{% endif %}
```

## Proposed change

Spread the arguments into the can method of the Timber\User class.

```php
	public function can( $capability, ...$args ) {
		return user_can( $this->ID, $capability, ...$args );
	}
```

## What version of WordPress, PHP and Timber are you using?
WordPress 5.8.3
PHP 7.4.27
Timber v2.14.7

## How did you install Timber? (for example, from GitHub, Composer/Packagist, WP.org?)
composer require timber/timber:^2.0

**Comment by gchtr** [comment]
Good catch! And thanks for taking the time set up such a clean issue. We should definitely pass those additional arguments down to `user_can()`.

**Comment by kshaner** [comment]
@gchtr what is the official php version support for Timber at this point? I was going to make the PR but the composer.json still specifies 5.3+ and the spread operator is 7.4+. If it's older than 7.4 I'll go ahead and do a call_user_func implementation.

**Comment by gchtr** [comment]
Hey @kshaner. It would be great if you could do a PR for this. Even if it’s without the tests. We can add those while reviewing the PR.

For Timber version 2, we will only support PHP 7.4+. For Timber version 1, we didn’t end support for version below 7.4 yet. We don’t want to invest too much time into 1.x, because we’re already thin on resources and want to make Timber v2 ready as soon as possible.

If you want to use the spread operator, I think we should add this to version 2 of Timber. You’ll need to work on the `2.x` branch instead of the `master` branch: see [https://github.com/timber/timber/blob/2.x/](https://github.com/timber/timber/blob/2.x/).

If you want to add it to Timber v1, then `call_user_func()` is the way to go.

**Comment by gchtr** [comment]
Resolved in #2632.

Thanks a lot for your work here, @kshaner 🤩
